src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -164,6 +164,22 @@ def _row_to_doc(row, kbase_collection, load_version, features, tool_genome_map,
     return doc
 
 
+def _create_tool_upa_map(tool_identifiers, source_dirs, meta_filenames):
```

Comment:
are you going to merge the `res` with `docs`? I think it's better to add the logic in `_row_to_doc`.  Also pandas dataframe is loaded into memory. I am not sure if this is a good idea to pass all meta_dfs.

Reply:
I already reverted back.